plugin/pulpcore/plugin/models/remote.py

```diff
@@ -34,28 +37,51 @@ def download_factory(self):
 
         Returns:
             DownloadFactory: The instantiated DownloaderFactory to be used by
-                get_downloader()
+                get_downloader().
         """
         try:
             return self._download_factory
         except AttributeError:
             self._download_factory = DownloaderFactory(self)
             return self._download_factory
 
-    def get_downloader(self, url, **kwargs):
+    def get_downloader(self, remote_artifact=None, url=None, **kwargs):
```

Comment:
`url` only is used for downloading the very-first metadata files from the remote where you don't yet have a RemoteArtifact for it since it's the first thing.